server/http_global_settings_handler_test.go

```diff
@@ -0,0 +1,191 @@
+package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/evcc-io/evcc/util"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSettingsSetJsonHandler(t *testing.T) {
+	type TestStruct struct {
+		Field1 string `json:"field1"`
+		Field2 int    `json:"field2"`
+	}
+
+	key := "testKey"
+	valueChan := make(chan util.Param, 1)
+
+	newStruc := func() any {
+		return &TestStruct{}
+	}
+
+	handler := settingsSetJsonHandler(key, valueChan, newStruc)
+
+	t.Run("valid JSON input", func(t *testing.T) {
+		input := TestStruct{
+			Field1: "value1",
+			Field2: 42,
+		}
+		body, _ := json.Marshal(input)
+
+		req := httptest.NewRequest(http.MethodPost, "/", bytes.NewReader(body))
+		w := httptest.NewRecorder()
+
+		handler(w, req)
+
+		resp := w.Result()
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+		select {
+		case param := <-valueChan:
+			assert.Equal(t, key, param.Key)
+			assert.Equal(t, &input, param.Val)
+		default:
+			t.Error("expected value to be sent to valueChan")
+		}
+	})
+
+	t.Run("invalid JSON input", func(t *testing.T) {
+		body := []byte(`{"field1": "value1", "field2": "invalid"}`)
+
+		req := httptest.NewRequest(http.MethodPost, "/", bytes.NewReader(body))
+		w := httptest.NewRecorder()
+
+		handler(w, req)
+
+		resp := w.Result()
+		assert.Equal(t, http.StatusBadRequest, resp.StatusCode)
+	})
+
+	t.Run("unknown fields in JSON input", func(t *testing.T) {
+		body := []byte(`{"field1": "value1", "field2": 42, "unknownField": "value"}`)
+
+		req := httptest.NewRequest(http.MethodPost, "/", bytes.NewReader(body))
+		w := httptest.NewRecorder()
+
+		handler(w, req)
+
+		resp := w.Result()
+		assert.Equal(t, http.StatusBadRequest, resp.StatusCode)
+	})
+}
+
+func TestMergeSettingsOld(t *testing.T) {
+
+	t.Run("old is nil", func(t *testing.T) {
```

Comment:
Einfacher wäre ein slice of Testcase mit alt/neu/ergebnis und Loop darüber, im Zweifel auch über structs.Map